capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java

```diff
@@ -128,4 +135,52 @@ public static void createUser(int userId, String userName, String email) throws
     }
     return itemTypes;
   }
+
+  /*
+   * Provided a list of ItemTypes, this method will return a list of Stores
+   * with every permutation of possible items from any given store that 
+   * satisfies this list of ItemTypes.
+   *
+   * @param itemTypes list of Strings that describes the desired Item Types
+   * @return a list of all potential stores and combinations that satisfy the Item Types
+   *
+   */
+  public static List<Store> getStoresWithItems(List<String> itemTypes) {
+    List<Store> stores = new ArrayList<Store>();
+    DatabaseClient dbClient = initClient();
+    String query = "";
+    boolean first = true;
+    for (String itemType : itemTypes) {
+      List<Store> newStores = new ArrayList<Store>();
+      query = "SELECT a.ItemId, a.ItemNameAndBrand, b.StoreId, b.Price, c.Address, c.StoreName " +
```

Comment:
Try using the `IN` clause? This still runs a query for every itemtype
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/bc813f9b21e360c3bc632f9b6d32f83686d46910/spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java#L1323